### PR TITLE
add strict type checking for tuple

### LINKED ISSUE #433: Add strict type checking in tuple
```python
def main():
  t: tuple[i32, str]
  t = (1, 2)

main()
```
This will generate an ASR, but actually should give an error that the second element is expected to be of `str` type.

### PR DESCRIPTION
**Pull request by yashsain**
Fixes #433

### CODE REVIEW

**Comment by certik** [comment]
Otherwise this looks good, I think that is how it should be done. Thanks for working on it.

File: src/libasr/asr_utils.h [hunk 1]
```diff
@@ -205,7 +205,8 @@ static inline std::string type_to_str_python(const ASR::ttype_t *t)
             return "str";
         }
         case ASR::ttypeType::Tuple: {
-            return "tuple";
+            ASR::Tuple_t *t = (ASR::Tuple_t *)t;
+            return "tuple[" + type_to_str_python(t->m_type) + "]";
```
[the remark this answers is not in the document]
Comment (yashsain) [reply src/libasr/asr_utils.h:209]: thanks for the review, I will try to make changes to it.

[the remark this answers is not in the document]
Comment (Thirumalai-Shaktivel) [reply src/libasr/asr_utils.h:209]: Yup, do something similar to this:
https://github.com/lcompilers/lpython/blob/ef633f2199fe70bb30dff17d3088a8b70e393250/src/libasr/asr_utils.cpp#L58-L59
replace `dependencies` with `type` and use `std::string` variable to store all the types information...

File: tests/errors/test_tuple1.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+def main():
+  t: tuple[i32, str]
+  t = (1, 2)
+
+main()
```
Comment (Thirumalai-Shaktivel) [inline tests/errors/test_tuple1.py:5]:
```suggestion
main()

```

File: tests/tests.toml [hunk 3]
```diff
@@ -410,3 +410,7 @@ asr = true
 [[test]]
 filename = "errors/test_dict5.py"
 asr = true
+
+[[test]]
+filename = "errors/test_tuple1.py"
+asr = true
```
Comment (Thirumalai-Shaktivel) [inline tests/tests.toml:416]:
```suggestion
asr = true

```

**Comment by Smit-create** [comment]
Ah, I mistakenly closed this with a forced update, not sure why? Opened #495 with @yashsain 's initial commit

**Pull request closed without merging**